File: 367. Valid Perfect Square.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        long long int l = 0, r = (int)(num/2 + 1);
        while (l <= r)
        {
            long long int mid = (l+r) >> 1;
            if (mid*mid == num) return true;
            else if (mid*mid < num)
                l = mid + 1;
            else
                r = mid - 1;
        }
        return false;
    }
};

File: 367. Valid Perfect Square.py
# the same as 69. Sqrt(x)
class Solution(object):
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        l = 0
        r = num/2 + 1
        while l <= r:
            mid = (l + r)/2
            if mid * mid == num:
                return True
            elif mid * mid < num:
                l = mid + 1
            else:
                r = mid - 1
        return False
